### ISSUE TITLE
root access required?

### ISSUE THREAD

**Issue by glyn** [tatsushid/go-fastping#8]
I get this error when I run the example code from the README in my own main or when I build the ping command from this project and invoke it thus:
```
$ ~/go/bin/ping www.example.com
Ping failed: listen ip4:icmp <nil>: operation not permitted
```
Running it under sudo succeeds however. Maybe it is worth adding a sentence to the README pointing out the need for root access.

**Comment by porjo** [comment]
`ip4:icmp` is a raw socket and so does require root privs. It is possible to implement ping as non-root on Linux and Darwin using `udp4` instead - see here for more info: https://godoc.org/golang.org/x/net/icmp#ListenPacket

There are some caveats - see here: http://stackoverflow.com/questions/8290046/icmp-sockets-linux/20105379#20105379

**Comment by tatsushid** [comment]
Yes now this implements only raw sock ping so root privs is required. I checked `net/icmp#ListenPacket` and I think there is no reason not to use it. I'll rewrite the code to use it instead of current `ListenIP`.

Also the README should be fixed.

**Comment by glyn** [comment]
Thanks!

On Thu, Jan 8, 2015 at 1:27 PM, Tatsushi Demachi <notifications@github.com>
wrote:

> Closed #8 <https://github.com/tatsushid/go-fastping/issues/8> via 8f74a61
> <https://github.com/tatsushid/go-fastping/commit/8f74a6128827dec687dad4906749d7ad0036dbca>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tatsushid/go-fastping/issues/8#event-215888191>.
>



-- 
Regards,
Glyn
